orchestrator/migrations/helpers.py

```diff
@@ -76,6 +76,177 @@ def create_missing_modify_note_workflows(conn):
             )
 
 
+def create_workflows(conn, new):
+    """
+    Create a new workflow
+
+    Args:
+        conn: DB connection as available in migration main file
+        new: an dict of your workflow data
+    
+    Example:
+        >>> new_workflows = {
+                "workflow_name": {
+                    "target": "CREATE",
+                    "description": "Workflow description",
+                    "tag": "ProductBlockName1",
+                    "search_phrase": "Search Phrase%",
+                }
+            }
+    """
+    for name, workflow in new.items():
+        workflow["name"] = name
+        conn.execute(
+            sa.text(
+                """
+                WITH new_workflow AS (
+                INSERT INTO workflows(name, target, description)
+                    VALUES (:name, :target, :description)
+                    RETURNING workflow_id)
+                INSERT
+                INTO products_workflows (product_id, workflow_id)
+                SELECT
+                p.product_id,
+                nw.workflow_id
+                FROM products AS p
+                    CROSS JOIN new_workflow AS nw
+                        WHERE p.tag = :tag
+                        AND p.name LIKE :search_phrase
+
+            """
+            ),
+            workflow,
+        )
+
+
+def create_fixed_inputs(conn, product_id, new):
+    """
+    Created fixed inputs for a product
+
+    Args:
+        conn: DB connection as available in migration main file
+        product_id: UUID of the product to link to
+        new: an dict of your workflow data
+        
+    Example:
+        >>> new = {
+                "fixed_input_1": "value",
+                "fixed_input_2": "value"
+            }
+    """
+    uuids = {}
+    for key, value in new.items():
+        uuids[key] = uuid4()
+        conn.execute(
+            sa.text(
+                "INSERT INTO fixed_inputs (fixed_input_id, name, value, created_at, product_id) "
+                "VALUES (:fixed_input_id, :key, :value, now(), :product_id)"
```

Comment:
You probably want to add `ON CONFLICT DO NOTHING` to make sure that if the fixed input already exists for some reason, that it wont fail the rest of the migration

Reply:
An extra design note:  currently we almost never, by convention, write a downgrade migration. Still not sure if that convention, will change in the future; but I can imagine it would be relatively easy to also generate the downgrade with these helpers.

E.g. removing a product/block; will also remove workflows

Reply:
@hmvp I have added the `ON CONFLICT DO NOTHING`, thanks. @acidjunk yes we may add downgrades in the future, good idea.